modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java

```diff
@@ -103,23 +92,53 @@ public static long addListTypeDefinition(
 			listTypeEntry -> listTypeEntries.put(
 				listTypeEntry.getKey(), listTypeEntry));
 
-		for (int i = 0; i < objectStatesJSONArray.length(); i++) {
-			JSONObject objectStateJSONObject =
-				objectStatesJSONArray.getJSONObject(i);
+		if (!ArrayUtil.isEmpty(stateFlowSettings)) {
+			JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+				JSONFactoryUtil.looseSerializeDeep(
+					stateFlowSettings[0].getValue()));
 
-			String key = objectStateJSONObject.getString("key");
+			JSONArray objectStatesJSONArray = jsonObject.getJSONArray(
+				"objectStates");
 
-			if (listTypeEntries.containsKey(key)) {
-				listTypeEntries.remove(key);
+			for (int i = 0; i < objectStatesJSONArray.length(); i++) {
+				JSONObject objectStateJSONObject =
+					objectStatesJSONArray.getJSONObject(i);
 
-				continue;
+				String key = objectStateJSONObject.getString("key");
+
+				if (listTypeEntries.containsKey(key)) {
+					listTypeEntries.remove(key);
+
+					continue;
+				}
+
+				listTypeEntryLocalService.addListTypeEntry(
+					null, userId, listTypeDefinition.getListTypeDefinitionId(),
+					key,
+					Collections.singletonMap(LocaleUtil.getDefault(), key));
 			}
+		}
 
-			listTypeEntryLocalService.addListTypeEntry(
-				null, userId, listTypeDefinition.getListTypeDefinitionId(), key,
-				Collections.singletonMap(LocaleUtil.getDefault(), key));
+		ObjectFieldSetting[] defaultValueSettings = ArrayUtil.filter(
```

Comment:
I think it is better to rename this variable to "defaultValueObjectFieldSettings". What do you think?

Reply:
Done!! :white_check_mark: